### PR TITLE
#431 rework the configsource api for clarification

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang6@googlemail.com>

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
  *******************************************************************************/
 package org.eclipse.microprofile.config.spi;
 
+import java.util.HashMap;
+import java.util.LinkedHashMap;
```
Comment (kenfinnigan) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: Don't think the `LinkedHashMap` import is used

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java [hunk 2]
```diff
@@ -81,7 +83,11 @@
...
      */
-    Map<String, String> getProperties();
+    default Map<String, String> getProperties() {
+        Map<String, String> props = new HashMap<>();
+        getPropertyNames().stream().forEach((prop) -> props.put(prop, getValue(prop)));
+        return props;
+    }
 
     /**
      * Gets all property names known to this config source, without evaluating the values.
```
Comment (ljnelson) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: I assume "without evaluating the values" should really be "potentially without evaluating the values".

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java (hunk 1 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: removed.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java (hunk 2 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: updated

**Pull request merged**